src/pages/home/report/ReactionList/PopoverReactionList.js

```diff
@@ -0,0 +1,222 @@
+            || this.state.popoverAnchorPosition !== nextState.popoverAnchorPosition
+            || previousLocale !== nextLocale;
+    }
+
+    componentWillUnmount() {
+        if (!this.dimensionsEventListener) {
+            return;
+        }
+        this.dimensionsEventListener.remove();
+    }
+
+    /**
+     * Get the PopoverReactionList anchor position
+     * We calculate the achor coordinates from measureInWindow async method
+     *
+     * @returns {Promise<Object>}
+     */
+    getReactionListMeasuredLocation() {
+        return new Promise((resolve) => {
+            if (this.reactionListAnchor) {
+                this.reactionListAnchor.measureInWindow((x, y) => resolve({x, y}));
+            } else {
+                resolve({x: 0, y: 0});
+            }
+        });
+    }
+
+    /**
+     * Show the ReactionList modal popover.
+     *
+     * @param {Object} [event] - A press event.
+     * @param {Element} reactionListAnchor - reactionListAnchor
+     * @param {Array} users - Array of users id
+     * @param {String} emojiName - Name of emoji
+     * @param {Array} emojiCodes - The emoji codes to display in the bubble.
+     * @param {Number} emojiCount - Count of emoji
+     * @param {Boolean} hasUserReacted - whether the current user has reacted to this emoji
+
+     */
+    showReactionList(
+        event,
+        reactionListAnchor,
+        users,
+        emojiName,
+        emojiCodes,
+        emojiCount,
+        hasUserReacted,
+    ) {
+        const nativeEvent = event.nativeEvent || {};
+
+        this.reactionListAnchor = reactionListAnchor;
+
+        // Singleton behaviour of ReactionListPopover creates race conditions when user requests multiple ReactionListPopover.
+        // But it is possible that every new request registers new callbacks thus instanceID is used to corelate those callbacks
+        this.instanceID = Math.random().toString(36).substr(2, 5);
+        this.getReactionListMeasuredLocation().then(({x, y}) => {
+            this.setState({
+                cursorRelativePosition: {
+                    horizontal: nativeEvent.pageX - x,
+                    vertical: nativeEvent.pageY - y,
+                },
+                popoverAnchorPosition: {
+                    horizontal: nativeEvent.pageX,
+                    vertical: nativeEvent.pageY,
+                },
+                users,
+                emojiName,
+                emojiCodes,
+                emojiCount,
+                isPopoverVisible: true,
+                hasUserReacted,
+            });
+        });
+    }
+
+    /**
+     * This gets called on Dimensions change to find the anchor coordinates for the action PopoverReactionList.
+     */
+    measureReactionListPosition() {
+        if (!this.state.isPopoverVisible) {
+            return;
+        }
+        this.getReactionListMeasuredLocation().then(({x, y}) => {
+            if (!x || !y) {
```

Comment:
this will also be the case when x or y can be `0`, which could be a valid case? Not sure if we want to add explicit null checks here.

Reply:
This is a valid case when we don't have an anchor. With zero values, we don't change the location of the popup during window resizing.
I can extend `getReactionListMeasuredLocation` with a failure scenario, but I have no problem with `measureInWindow` so far

Reply:
kk, then leave it just like it is 👍